web-console/src/components/auto-form.tsx

```diff
@@ -16,26 +16,29 @@
  * limitations under the License.
  */
 
-import { resolveSrv } from 'dns';
 import * as React from 'react';
-import axios from 'axios';
 import { InputGroup } from "@blueprintjs/core";
-import { HTMLSelect, FormGroup, NumericInput, TagInput } from "../components/filler";
+import { HTMLSelect, FormGroup, NumericInput, TagInput, JSONInput } from "../components/filler";
+import "./auto-form.scss";
+import {parseStringToJSON, validJson} from "../utils";
 
 interface Field {
   name: string;
   label?: string;
-  type: 'number' | 'size-bytes' | 'string' | 'boolean' | 'string-array';
+  type: 'number' | 'size-bytes' | 'string' | 'boolean' | 'string-array' | 'json';
   min?: number;
 }
 
 export interface AutoFormProps<T> extends React.Props<any> {
   fields: Field[];
   model: T | null,
-  onChange: (newValue: T) => void
+  onChange: (newValue: T) => void,
+  JSONErrors?: T | null,
```

Comment:
do not start property names with capital letters. Treat `json` just like a normal word in the dictionary for property names.